go-controller/pkg/allocator/pod/macs.go

```diff
@@ -0,0 +1,157 @@
+package pod
+
+import (
+	"fmt"
+	"net"
+
+	ipamclaimsapi "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+	k8snet "k8s.io/utils/net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+// GetMACOwner compose the owner identifier reserved for MAC addresses management.
+// Returns "<ns>/<pod-name>" for regular pods and "<ns>/<vm-name>" for VMs.
+func GetMACOwner(pod *corev1.Pod) string {
+	// Check if this is a VM pod and persistent IPs are enabled
+	if vmName, ok := pod.Labels[kubevirtv1.VirtualMachineNameLabel]; ok {
+		return fmt.Sprintf("%s/%s", pod.Namespace, vmName)
+	}
+
+	// Default to pod-based identifier
+	return fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
+}
+
+// ReleasePodReservedMacAddress releases pod's reserved MAC address, if exists.
+// It removes the used MAC address, from pod network annotation, and remove it from the MAC manager store.
+func (allocator *PodAnnotationAllocator) ReleasePodReservedMacAddress(pod *corev1.Pod, mac net.HardwareAddr, claim *ipamclaimsapi.IPAMClaim) {
+	networkName := allocator.netInfo.GetNetworkName()
+	if allocator.macManager == nil {
+		klog.V(5).Infof("No mac manager for network %s, skipping mac address release", networkName)
+		return
+	}
+
+	if _, exist := pod.Labels[kubevirtv1.VirtualMachineNameLabel]; exist && claim != nil {
+		klog.V(5).Infof("Retaining MAC address (%s) of pod (%s/%s) on network (%s) becuase it has IPAM claim (%s/%s)",
```

Comment:
```suggestion
		klog.V(5).Infof("Retaining MAC address %q of pod \"%s/%s\" on network %q because it has IPAM claim \"%s/%s\"",
```

Reply:
IIUC, you're using the IPAMClaim as the artifact that represents the VM - i.e. while there's an IPAMClaim around for VM XYZ, the MAC address for it will *not* be released. 

This makes sense to me, but I wonder about the log message above. You print the POD name. Would it make sense to print instead the VM name ?

Reply:
I think we should only do this for VMs. I don't know if ovn-k will ever support ipam claims for anything other than vms, but, when it does, I would rather have these tests not failing. 

IMHO, this should happen only for VM pods. 

I am good if you insist though.